controllers/numaresourcesoperator_controller.go

```diff
@@ -164,18 +164,27 @@ func (r *NUMAResourcesOperatorReconciler) Reconcile(ctx context.Context, req ctr
 	}
 
 	result, condition, err := r.reconcileResource(ctx, instance, trees)
-	if condition != "" {
-		if condition == status.ConditionAvailable && multiMCPsErr != nil {
-			_, _ = r.updateStatus(ctx, instance, status.ConditionDegraded, validation.NodeGroupsError, multiMCPsErr)
-		} else {
-			_, _ = r.updateStatus(ctx, instance, condition, reasonFromError(err), err)
-		}
+
+	_ = updateStatusConditionsIfNeeded(instance, condition, reasonFromError(err), messageFromError(err))
```

Comment:
we need to ensure we don't lose the error data when parsing the message. this is related to how the errors are built, so unwrapping a non-nil error can result in a nil error which then will cause the main reason of the degraded state to be lost. We can double verify that and handle this in a cleanup PR

Reply:
I'll verify and post the needed changes

Reply:
we can probably fix this by fixing `messageFromError` and `reasonFromError`. I'd be very interested in learning on which cases we can lose errors.

Reply:
thanks for addressing